src/builder/modules/Generator.res

```diff
@@ -156,6 +226,53 @@ let drawFoldLine = (from: Builder.position, to: Builder.position) => {
     )
 }
 
+let drawFoldLinePath = (points: array<Builder.position>, ~close: bool=false, ()) => {
+  model :=
+    Generator_Builder.drawPath(
+      model.contents,
+      points,
+      ~color="#7b7b7b",
+      ~width=1,
+      ~pattern=[2, 2],
+      ~offset=3,
+      ~close,
+    )
+}
+
+let drawFoldLineRect = (dest: Builder.rectangle) => {
+  let (x, y, w, h) = dest
+  model :=
+    Generator_Builder.drawPath(
+      model.contents,
+      [(x, y), (x + w, y), (x + w, y + h), (x - 1, y + h), (x - 1, y)],
+      ~color="#7b7b7b",
+      ~width=1,
+      ~pattern=[2, 2],
+      ~offset=3,
+      ~close=false,
+    )
+}
+
+let drawFoldLineCuboid = (
```

Comment:
This one is a very interesting idea.

Your recent suggestions have led me to also think about having a `drawTab()` function.

This also leads me to think about where the fold lines belong.

For example, it could be something like this:

[1] `drawFoldLineCuboid()` only draws the internal fold lines, and no edge/external fold lines.
[2] `drawTab()` draws a fold line at its base, which then becomes an edge fold on a cuboid.

These days I am wondering about the value of having fold lines along the outsides of designs, since no fold is actually needed there.

Let me know your thoughts on the above.

Reply:
Some of the exterior "fold" lines actually are still folded along even if they are not the border of two faces- they are still folded when the tabs are folded. The rest of the exterior folds on the cuboid still represent *where* a fold should be, and though they aren't where a fold is actually made, they can help with aligning where the edges of the paper are.

drawTab() is something that would be great thing to have, but I want to see if the antialiasing issue can be dealt with first before adding it. As for having the fold for that edge be part of the drawTav() function, this would probably add unnecessary complexity, as most generators have toggleable folds, meaning the function would also need a boolean for whether to draw one or not. In a case like this, ~drawFolds and ~drawTabs parameters might as well be added to the drawCuboid() function instead, so that the texture, tabs and folds could be drawn in one step.

Reply:
> Some of the exterior "fold" lines actually are still folded along even if they are not the border of two faces- they are still folded when the tabs are folded.

👍 

> The rest of the exterior folds on the cuboid still represent where a fold should be, and though they aren't where a fold is actually made, they can help with aligning where the edges of the paper are.

Can you explain a bit more about the aligning idea?

> drawTab() is something that would be great thing to have, but I want to see if the antialiasing issue can be dealt with first before adding it.

TBH I am not concerned with the antialiasing here. If I had to choose between generated antialiased lines in automatic vs pixel perfect tabs in images used as overlays, I would definitely favour the generated version - it's a great improvement over the image strategy.

We could implement our own line drawing function which is not antialiased, as you can image, it can also look a bit rough depending on the angle.

> As for having the fold for that edge be part of the drawTav() function.

You've raised some good points here.

There are some implementation details to work out, but here's the concept I have in mind - we only draw a fold line where you literally need to do a fold. We don't draw fold lines on edges where there are no tabs because there is no fold (i.e. a join is not a fold).

This is pretty easy to implement, and should produce a nice consistent pattern we can apply to all generators. What do you think?

Reply:
Having outlines even where there isn't a fold has a few benefits over not having them:

- It aids with cutting and aligning edges- When cutting along the very edge of a face, a dashed line can help with making a straighter cut. It gives both an easily visible and more forgiving border that can be seen better than the border between the blank and textured parts of the paper, especially when the paper is farther from view if it is cut with a knife on a desk, or when the texture is white or close to white. When gluing edge to a tab, this visual advantage remains, and can both help with making a cuboid as straight as possible, and with making any slight errors less obvious.
- It makes designing with different tab placements easier- Since sometimes tabs need to be put on edges other than where they are usually, if outside folds are only on tabs, then this adds complexity to where they are drawn. drawFoldLineCuboid() only draws two rectangles and a line and draws on every edge, but not drawing on bare edges increases this to 6 calls of drawFoldLine() at the simplest, and needing to account for whether a tab is present means individually drawing along every edge, defeating the function's purpose. It would be much simpler to treat every edge the same.
- Having fold lines as edge lines is also a much more common standard to be used than having just the folded edges given dashed lines, especially after solid red outlines fell out of favor because of their aesthetics. As opposed to adjusting or creating new standards for designs, I think that the generator functions should instead be to perpetuate and enable existing conventions, making them more attractive to use and discouraging designers from creating new ones that conflict with existing ones.

Reply:
Some great points here.

Once we make a call on what we want the end result to look like, it will help guide decisions on what functions we need and how they should behave.